### PR TITLE
refactor(webpack): add ability to get webpack config class

### PR DESCRIPTION
**Pull request by damianstasik**
## Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
This PR is an improvement of #5366 and adds a new method to `WebpackBundler` that returns a webpack config class (`WebpackClientConfig`, `WebpackServerConfig` or `WebpackModernConfig`).

I also cleaned up the `build` method a bit and moved the plugin aliases into `WebpackClientConfig` and `WebpackServerConfig` so that each class has all of it's webpack aliases in one place.

## Checklist:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Comment by galvez** [comment]
@visualfanatic Please ensure the test suite passes first.

**Comment by visualfanatic** [comment]
@galvez @pi0 fixed, could you take a look?

File: packages/webpack/src/builder.js [hunk 1]
```diff
@@ -37,44 +37,32 @@ export class WebpackBundler {
     }
   }
 
+  getWebpackConfig(name) {
```
Comment (clarkdo) [inline packages/webpack/src/builder.js]: How about returning `config` object here, then we don't need an extract `bind`

**Comment by clarkdo** [comment]
Changed title to `refactor` for being eligible for v2.5.2

File: packages/webpack/src/builder.js (hunk 1 above)
Comment (pi0) [reply packages/webpack/src/builder.js]: @clarkdo Can you please elaborate more about bind?

This function is not async to leverage `Promise.all`. The runner will be specified based on build type [here](https://github.com/nuxt/nuxt.js/blob/82abc2632f8da8343d32128377b654838b963f51/packages/webpack/src/builder.js#L106)

Comment (pi0) [reply packages/webpack/src/builder.js]: @clarkdo Refactor applied. Please review again.

Comment (visualfanatic) [reply packages/webpack/src/builder.js]: @pi0 your change was a surprise, but it was appreciated, thank you.

**Comment by juliendargelos** [comment]
As I proposed [here](https://github.com/nuxt/nuxt.js/pull/5425), I think it could be useful for some advanced modules to be able to extend the webpack base config class for independent webpack build based on Nuxt base config. It's just about exposing [WebpackBaseConfig](https://github.com/visualfanatic/nuxt.js/blob/5c1e540747067c6a938acbe812dd6624767e265b/packages/webpack/src/config/base.js) (and possibly `WebpackClientConfig`, `WebpackServerConfig` and `WebpackModernConfig`) in [index.js](https://github.com/visualfanatic/nuxt.js/blob/5c1e540747067c6a938acbe812dd6624767e265b/packages/webpack/src/index.js):

```javascript
export { WebpackBundler as BundleBuilder } from './builder'
export { default as WebpackBaseConfig } from './config/base'
```

**Comment by pi0** [comment]
There is no guarantee about structure of webpack config classes to be the same across releases. So one can use final result and use/extend it for advanced use cases.

**Pull request merged**